plutus-playground/plutus-playground-server/usecases/Game.hs

```diff
@@ -1,54 +1,54 @@
 module Language.PlutusTx.Coordination.Contracts.Game where
 
 import qualified Language.PlutusTx            as PlutusTx
+import qualified Language.PlutusTx.Prelude    as P
 import           Ledger
 import           Ledger.Validation
 import           Wallet
 import           Playground.Contract
 
-logAMessage :: MockWallet ()
-logAMessage = logMsg "wallet log"
+import qualified Data.ByteString.Char8        as C
 
-data ANumber = ANumber Int
+data HashedString = HashedString ByteString
     deriving (Generic, ToJSON, FromJSON, ToSchema)
 
-PlutusTx.makeLift ''ANumber
+PlutusTx.makeLift ''HashedString
 
-data AGuess = AGuess Int
+data ClearString = ClearString ByteString
     deriving (Generic, ToJSON, FromJSON, ToSchema)
 
-PlutusTx.makeLift ''AGuess
+PlutusTx.makeLift ''ClearString
 
 gameValidator :: ValidatorScript
 gameValidator = ValidatorScript (Ledger.fromPlcCode $$(PlutusTx.plutus [||
-    \(AGuess guess) (ANumber number) (p :: PendingTx ValidatorHash) ->
+    \(ClearString guess) (HashedString actual) (p :: PendingTx ValidatorHash) ->
 
-    if guess == number
+    if $$(P.equalsByteString) actual ($$(P.sha2_256) guess)
     then ()
-    else $$(PlutusTx.traceH) "WRONG!" ($$(PlutusTx.error) ())
+    else $$(P.traceH) "WRONG!" ($$(P.error) ())
 
     ||]))
 
 gameAddress :: Address'
 gameAddress = Ledger.scriptAddress gameValidator
 
-contribute :: (WalletAPI m, WalletDiagnostics m)
-    => Int
-    -> Value
-    -> m ()
-contribute n value = do
-    let ds = DataScript (Ledger.lifted (ANumber n))
+lock :: String -> Value -> MockWallet ()
+lock word value = do
+    let hashedWord = plcSHA2_256 (C.pack word)
+        ds = DataScript (Ledger.lifted (HashedString hashedWord))
```

Comment:
could you perhaps move `HashedString (plcSHA_256 C.pack string))` into a top level function, along with `clearWord` perhaps, I think it's a bit confusing having all the wrapping, packing and hashing inside the "logic" function.